internal/node/hybrid/certificate_validator.go

```diff
@@ -0,0 +1,178 @@
+package hybrid
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+)
+
+// CertificateType represents the type of certificate being validated
+type CertificateType string
+
+const (
+	CertTypeKubelet CertificateType = "kubelet"            // kubelet certificate
+	CertTypeIAMRA   CertificateType = "iam-roles-anywhere" // IAM Roles Anywhere certificate
+)
+
+// ValidationError represents a kubelet certificate validation error with remediation information
+type ValidationError struct {
+	message          string
+	remediation      string
+	cause            error
+	isDateValidation bool
+	noCert           bool
+}
+
+type RemediationMessages struct {
+	noCert            string
+	invalidFormat     string
+	clockSkewDetected string
+	expired           string
+	invalidCA         string
+}
+
+func (e *ValidationError) Error() string {
+	if e.cause != nil {
+		return fmt.Sprintf("%s: %v", e.message, e.cause)
+	}
+	return e.message
+}
+
+func (e *ValidationError) Unwrap() error {
+	return e.cause
+}
+
+func (e *ValidationError) Remediation() string {
+	return e.remediation
+}
+
+func IsDateValidationError(err error) bool {
+	var v *ValidationError
+	return errors.As(err, &v) && v.isDateValidation
+}
+
+func IsNoCertError(err error) bool {
+	var v *ValidationError
+	return errors.As(err, &v) && v.noCert
+}
+
+func getRemediationMessage(certPath string, certType CertificateType) RemediationMessages {
```

Comment:
should we just return the remediation as an enum and then let the callers, the kubelet validator and iam-ra validator generate their own remediation messages?  basically what you did here, just split out and push up the stack